controllers/appmesh/cloudmap_controller.go

```diff
@@ -36,16 +35,23 @@ type cloudMapReconciler struct {
 	finalizerManager            k8s.FinalizerManager
 	cloudMapResourceManager     cloudmap.ResourceManager
 	enqueueRequestsForPodEvents handler.EventHandler
+	notificationChannel         chan interface{}
```

Comment:
it's better to have three type-safe channel than a single non-type safe channel. 
explicit is always better than implicit

Reply:
the channel declaration should be `chan <- "the type"` on receiver side.